frontend/src/services/blockServices.js

```diff
@@ -0,0 +1,90 @@
+export const getAllBlocks = async () => {
+  const URL = `${import.meta.env.VITE_BACKEND_URL}/api/v1/blocks`;
+
+  try {
+    const res = await fetch(URL);
+    const data = await res.json();
+
+    if (!res.ok) {
+      throw new Error(data.message || 'Failed to fetch blocks');
+    }
+
+    return data.payload;
+  } catch (error) {
+    throw new Error(error.message || 'Failed to fetch blocks');
+  }
+};
+
+export const getBlock = async () => {
+  const URL = `${import.meta.env.VITE_BACKEND_URL}/api/v1/blocks`;
+
+  try {
+    const res = await fetch(URL);
+    const data = await res.json();
+
+    if (!res.ok) {
+      throw new Error(data.message || 'Failed to fetch block');
+    }
+
+    return data.payload;
+  } catch (error) {
+    throw new Error(error.message || 'Failed to fetch block');
+  }
+};
+
+export const getBlockByHash = async (hash) => {
+  if (!hash) {
+    throw new Error('Block hash is required');
+  }
+  const URL = `${import.meta.env.VITE_BACKEND_URL}/api/v1/blocks/${hash}`;
```

Comment:
Add ` hash/ ` in-between ` blocks/ ` and ` ${hash} ` to get the correct endpoint here.